### PR TITLE
adding a script to fetch and convert devin's output for evaluation

### PR DESCRIPTION
**Pull request by Jiaxin-Pei**

### CODE REVIEW

File: evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py
Comment (xingyaoww) [inline evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py:1]: How about we put this file to `SWE-Bench/scripts`?

File: evaluation/README.md [hunk 1]
```diff
@@ -11,5 +11,8 @@ all the preprocessing/evaluation/analysis scripts.
...
 ## Tasks
 ### SWE-bench
-- analysis
-  - devin_eval_analysis.ipynb: notebook analyzing devin's outputs
\ No newline at end of file
+- notebooks
+  - `devin_eval_analysis.ipynb`: notebook analyzing devin's outputs
+- src
+  - `prepare_devin_outputs_for_evaluation.py`: script fetching and converting devin's output into the desired json file for evaluation.
+    - outputs: two json files under `evaluation/SWE-bench/data/` that can be directly used for evaluation
```
[the remark this answers is not in the document]
Comment (Jiaxin-Pei) [reply evaluation/README.md:18]: requested

File: evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    with open(os.path.join(output_dir, "pass_output.json"), "w") as pass_file:
+        json.dump(pass_files_info, pass_file, indent=4)
+
+    with open(os.path.join(output_dir, "fail_output.json"), "w") as fail_file:
+        json.dump(failed_files_info, fail_file, indent=4)
+
```
[the remark this answers is not in the document]
Comment (Jiaxin-Pei) [reply evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py:59]: It only takes ~1 minute to fetch and process the files. The purpose of having two files is you can directly start from the passed files for pilot testing. I can generate another merged file and upload it to HF

[the remark this answers is not in the document]
Comment (xingyaoww) [reply evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py:59]: having both options is a good option! maybe we can add an argument in the script to switch that behavior; and we can upload both version to HF and have user decide which one they want to download

[the remark this answers is not in the document]
Comment (Jiaxin-Pei) [reply evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py:59]: sounds good!

File: evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py
Comment (Jiaxin-Pei) [reply evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py:1]: I'm not quite sure about this. It seems more reasonable to keep dataset related files in the dataset folder to me. @JustinLin610 @libowen2121 any thoughts on this?

Comment (xingyaoww) [reply evaluation/SWE-bench/src/prepare_devin_outputs_for_evaluation.py:1]: Ohh i suggest we do this: `mv src/prepare_devin_outputs_for_evaluation.py scripts/prepare_devin_outputs_for_evaluation.py`

**Review by xingyaoww — APPROVED** [review]
LGTM!

**Pull request merged**